cypress/integration/x-msc2716-historical-import/historical-import.spec.ts

```diff
@@ -0,0 +1,288 @@
+            },
+            // We'll handle the errors ourselves below
+            failOnStatusCode: false,
+        })
+            .then((res) => {
+                // Registration success
+                if(res.status === 200) {
+                    return;
+                }
+
+                const errcode = res.body.errcode;
+
+                // User already registered and good to go
+                if (res.status == 400 && errcode === "M_USER_IN_USE") {
+                    return;
+                }
+
+                const errorMessage = res.body.error;
+                throw new Error(`ensureVirtualUserRegistered failed to register: (${errcode}) ${errorMessage}`)
+            });
+    });
+}
+
+describe("MSC2716: Historical Import", () => {
+    let synapse: SynapseInstance;
+    let asMatrixClient: MatrixClient;
+    let loggedInUserCreds: UserCredentials;
+    const virtualUserIDs = ['@maria-01234:localhost'];
+
+    // This corresponds to the application service registration defined in the
+    // "msc2716-historical-import" Synapse configuration
+    const AS_TOKEN = 'as_token123';
+
+    beforeEach(() => {
+        cy.window().then(win => {
+            // Collapse left panel for these tests (get more space in the area we care about)
+            win.localStorage.setItem("mx_lhs_size", "0");
+        });
+        // Start Synapse with msc2716_enabled and an application service configured
+        cy.startSynapse("msc2716-historical-import").then(data => {
+            synapse = data;
+
+            // This is the person we're logged in as
+            cy.initTestUser(synapse, "Grace").then(userCreds => {
+                loggedInUserCreds = userCreds;
+            });
+            // After the page is loaded from initializing the logged in user so
+            // the mxSettingsStore is available
+            cy.window().then(win => {
+                // Disable the sound notifications so you're not startled and
+                // confused where the sound is coming from
+                win.mxSettingsStore.setValue("audioNotificationsEnabled", null, SettingLevel.DEVICE, false);
+                // Show hidden events to make debugging easier. And the state
+                // events will show up in the timeline to make it the same
+                // assert as other events.
+                win.mxSettingsStore.setValue("showHiddenEventsInTimeline", null, SettingLevel.DEVICE, true);
+            });
+
+            // Get a Matrix Client for the application service
+            cy.newMatrixClient(synapse, {
+                userId: '@gitter-badger:localhost',
+                accessToken: AS_TOKEN,
+            }).then(matrixClient => {
+                asMatrixClient = matrixClient;
+            });
+
+            ensureVirtualUsersRegistered(synapse, AS_TOKEN, virtualUserIDs);
+        });
+    });
+
+    afterEach(() => {
+        cy.stopSynapse(synapse);
+    });
+
+    it("asdf", () => {
```

Comment:
Working test and looks reasonably as expected ✅

<img width="897" alt="" src="https://user-images.githubusercontent.com/558581/169641735-26910659-18e7-4d5c-a3e2-5a62b6640569.png">